### PR TITLE
Weight dropout

### PR DESCRIPTION
**Pull request by Judyxujj**
This PR adds the weight dropout

### CODE REVIEW

**Comment by Judyxujj** [comment]
> I think following `torch.nn.utils.weight_norm` or `torch.nn.utils.parametrizations.weight_norm` for this makes more sense, or not? This doesn't need to be (or rather should not be) specific for one module (here it is specific to `torch.nn.Linear`) but it should be generic, and less hacky, and simpler.

**Comment by Judyxujj** [comment]
> I think following `torch.nn.utils.weight_norm` or `torch.nn.utils.parametrizations.weight_norm` for this makes more sense, or not? This doesn't need to be (or rather should not be) specific for one module (here it is specific to `torch.nn.Linear`) but it should be generic, and less hacky, and simpler.

But here we can also use the WeightDrop class to wrap other torch.nn.Module. I add

**Comment by curufinwe** [comment]
No longer needed. If we are interested again we can use the pytorch functions that Albert mentioned.
